### PR TITLE
get args fix

### LINKED ISSUE #1087: Crash when logging `global`
### Version

0.1.5

### Platform

Linux {device-name} 5.15.0-43-generic #46-Ubuntu SMP Tue Jul 12 10:30:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### What steps will reproduce the bug?

```js
console.log(global)
```
then run `bun <file>`

### How often does it reproduce? Is there a required condition?

_No response_

### What is the expected behavior?

_No response_

### What do you see instead?

> Please take down the issue, if there are sensitive information's!

```shell
Detected offset inconsistency: inlineOverflowAccordingToTotalSize doesn't match numberOfOutOfLineSlotsForMaxOffset!
this = 0x7f6d0001b560
transitionOffset = -1
maxOffset = 64
m_inlineCapacity = 75
propertyTable = 0x7f6ec7427000
numberOfSlotsForMaxOffset = 65
totalSize = 65
inlineOverflowAccordingToTotalSize = 0
numberOfOutOfLineSlotsForMaxOffset = 1

Crash at 4294967309674


–––– bun meta ––––
Bun v0.1.5 Linux x64 #46-Ubuntu SMP Tue Jul 12 10:30:17 UTC 2022
AutoCommand: bun_bun 
Elapsed: 30ms | User: 27ms | Sys: 3ms
RSS: 67.11MB | Peak: 30.73MB | Commit: 67.11MB | Faults: 0
–––– bun meta ––––

Ask for #help in https://bun.sh/discord or go to https://bun.sh/issues

```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by dylan-conway**
fixes #1087

### CODE REVIEW

File: test/bun.js/print-process-args.js [hunk 1]
```diff
@@ -0,0 +1 @@
+console.log(process.argv);
```
Comment (Jarred-Sumner) [inline test/bun.js/print-process-args.js]: why not `JSON.stringify`? 

also we can avoid console.log here and just do 

```js
var writer = Bun.stdout.writer()
writer.write(JSON.stringify(process.argv));
await writer.flush(true);
process.exit(0);
```

File: test/bun.js/process-args.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { test, expect } from "bun:test";
+
+test("args exclude run", async () => {
+    const arg_prefix = '[ "' + process.argv[0] + '", "' + import.meta.dir + '/print-process-args.js"';
+
+    const { stdout: s1 } = spawn(["bun-debug", "print-process-args.js"], { cwd: import.meta.dir });
+    const t1 = await (await new Response(s1).text()).trim();
+    const e1 = new String(arg_prefix + ' ]\n').trim();
+    expect(t1).toBe(e1);
+    console.log(t1.length + ": " + t1);
```
Comment (Jarred-Sumner) [inline test/bun.js/process-args.test.js]: JSON.parse instead of parsing the console.log output

File: src/cli/test_command.zig [hunk 3]
```diff
@@ -327,7 +327,7 @@ pub const TestCommand = struct {
         js_ast.Expr.Data.Store.create(default_allocator);
         js_ast.Stmt.Data.Store.create(default_allocator);
         var vm = try JSC.VirtualMachine.init(ctx.allocator, ctx.args, null, ctx.log, env_loader);
-        vm.argv = ctx.positionals;
+        vm.argv = ctx.passthrough;
```
Comment (Jarred-Sumner) [inline src/cli/test_command.zig:330]: we should also test that this one works via running `bun wiptest`

Comment (Jarred-Sumner) [reply src/cli/test_command.zig:330]: but if it doesnt we should merge this without that change

Comment (dylan-conway) [reply src/cli/test_command.zig:330]: `bun-debug wiptest process-args.test.js` is okay, but `bun-debug wiptest` isn't happy with the json parsing:
```
const t1 = JSON.parse(await new Response(s1).text());
                   ^
SyntaxError: JSON Parse error: Unexpected EOF
      at /Users/dylan/code/bun/test/bun.js/process-args.test.js:12:15
```

Comment (dylan-conway) [reply src/cli/test_command.zig:330]: nevermind, it is working with `bun-debug wiptest`. I ran it from the wrong directory and there are other error that got in the way

**Pull request merged**